src/boardState.ts

```diff
@@ -0,0 +1,239 @@
+
+        for(var i=0; i<shipLengths.length; i++){
+            if(numberRemaining[shipLengths[i]]>0){
+                let nextShip: ShipPlacement;
+
+
+                if(validShipPositions[i].numberOfPossibilities>0){
+                    if(!exhaustive){
+                        nextShip = validShipPositions[i].pickRandomPlacement();
+                    }
+                    else{
+                        let nextStep: number = 0;
+                        if(pathToFollow.length > currentPathIndex){
+                            nextStep = pathToFollow[currentPathIndex];
+                            if(nextStep > validShipPositions[i].numberOfPossibilities-1){
+                                let nextPath:number[] = currentPath;
+                                nextPath[nextPath.length-1]++;
+                                return {state: new BoardState([]),found:false, nextPath: nextPath};
+                            }
+                        }
+                        currentPathIndex++;
+                        currentPath.push(nextStep);
+                        nextShip = validShipPositions[i].pickSpecificPlacement(nextStep);
+                    }
+                }
+                else{
+                    if(!exhaustive)
+                    {
+                        return {state: new BoardState([]), found:false, nextPath: []};
+                    }else{
+                        let nextPath:number[] = currentPath;
+                        nextPath[nextPath.length-1]++;
+                        return {state: new BoardState([]),found:false, nextPath: nextPath};
+                    }
+                }
+                
+                for(var j=0; j<validShipPositions.length; j++){
+                    validShipPositions[j].removePossibilities(nextShip);
+                }
+                possibleShips.push(nextShip);
+                numberRemaining[shipLengths[i]]--;
+            }
+        }
+        let nextPath:number[] = currentPath;
+        nextPath[nextPath.length-1]++;
+        return {state: new BoardState(possibleShips), found:true, nextPath}
+
+        //pick first shot
+        //choose
+    }
+
+    static getRandomSet(shipLengths: Array<number>, shots:Array<Shot> = [], hitShots: Array<Shot> = []):{state:BoardState, doSwitch:boolean, nextPath: number[]}{
+        let returnValue: {state: BoardState, nextPath:number[], found: boolean};
+        let tries:number =0;
+        do{
+            tries++;
+            returnValue = BoardState.tryGetRandomSet(shipLengths, shots, hitShots);
+            if(tries>5){
+                return {state: returnValue.state, doSwitch:true, nextPath: []}
+            }
+        }while(!returnValue.found)
+        return {state: returnValue.state, doSwitch:false, nextPath: []};
+    }
+    
+
+    public drawShips(){
+        let board:Array<Array<boolean>> = this.getBoardArray();
+
+        console.log("ships:");
+        
+        for(var i=0;i <Globals.BOARD_ROWS; i++){
```

Comment:
Really nice use of a config file!

I'm not sure I'd call it `Globals` - that has further connotations - but yeah, I like the setup.